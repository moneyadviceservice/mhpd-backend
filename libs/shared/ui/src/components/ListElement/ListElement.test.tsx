import React, { ReactNode } from 'react';
import { render, screen } from '@testing-library/react';
import { ListElement } from '.';

describe('List component', () => {
  const Content: ReactNode[] = [];
  beforeAll(() => {
    Content.push(<p>First line</p>);
    Content.push(<p>Second Line</p>);
    Content.push(<p>Third Line</p>);
  });
  it('renders the unordered magenta list correctly', () => {
    render(<ListElement items={Content} color="magenta" variant="unordered" />);
    const list = screen.getByTestId('list-element');
    expect(list).toMatchSnapshot();
  });

  it('renders the unordered dark list correctly', () => {
    render(<ListElement items={Content} color="dark" variant="unordered" />);
    const list = screen.getByTestId('list-element');
    expect(list).toMatchSnapshot();
  });

  it('renders the unordered blue list correctly', () => {
    render(<ListElement items={Content} color="blue" variant="unordered" />);
    const list = screen.getByTestId('list-element');
    expect(list).toMatchSnapshot();
  });

  it('renders the unordered magenta list correctly', () => {
    render(<ListElement items={Content} color="magenta" variant="ordered" />);
    const list = screen.getByTestId('list-element');
    expect(list).toMatchSnapshot();
  });

  it('renders the unordered dark list correctly', () => {
    render(<ListElement items={Content} color="dark" variant="ordered" />);
    const list = screen.getByTestId('list-element');
    expect(list).toMatchSnapshot();
  });

  it('renders the unordered blue list correctly', () => {
    render(<ListElement items={Content} color="blue" variant="ordered" />);
    const list = screen.getByTestId('list-element');
    expect(list).toMatchSnapshot();
  });
});
