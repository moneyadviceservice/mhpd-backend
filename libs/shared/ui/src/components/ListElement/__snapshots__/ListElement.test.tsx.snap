// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders the unordered blue list correctly 1`] = `
<div
  class="flex items-center"
  data-testid="list-element"
>
  <ul
    class="list-disc marker:text-blue-800 marker:mr-2 marker:pr-2 space-y-2 marker:leading-snug"
    start="-1"
  >
    <li>
      <p>
        First line
      </p>
    </li>
    <li>
      <p>
        Second Line
      </p>
    </li>
    <li>
      <p>
        Third Line
      </p>
    </li>
  </ul>
</div>
`;

exports[`List component renders the unordered blue list correctly 2`] = `
<div
  class="flex items-center"
  data-testid="list-element"
>
  <ol
    class="list-decimal marker:text-blue-800 marker:mr-2 marker:pr-2 space-y-2 marker:leading-snug"
  >
    <li>
      <p>
        First line
      </p>
    </li>
    <li>
      <p>
        Second Line
      </p>
    </li>
    <li>
      <p>
        Third Line
      </p>
    </li>
  </ol>
</div>
`;

exports[`List component renders the unordered dark list correctly 1`] = `
<div
  class="flex items-center"
  data-testid="list-element"
>
  <ul
    class="list-disc marker:mr-2 marker:pr-2 space-y-2 marker:leading-snug"
    start="-1"
  >
    <li>
      <p>
        First line
      </p>
    </li>
    <li>
      <p>
        Second Line
      </p>
    </li>
    <li>
      <p>
        Third Line
      </p>
    </li>
  </ul>
</div>
`;

exports[`List component renders the unordered dark list correctly 2`] = `
<div
  class="flex items-center"
  data-testid="list-element"
>
  <ol
    class="list-decimal marker:mr-2 marker:pr-2 space-y-2 marker:leading-snug"
  >
    <li>
      <p>
        First line
      </p>
    </li>
    <li>
      <p>
        Second Line
      </p>
    </li>
    <li>
      <p>
        Third Line
      </p>
    </li>
  </ol>
</div>
`;

exports[`List component renders the unordered magenta list correctly 1`] = `
<div
  class="flex items-center"
  data-testid="list-element"
>
  <ul
    class="list-disc marker:text-pink-800 marker:mr-2 marker:pr-2 space-y-2 marker:leading-snug"
    start="-1"
  >
    <li>
      <p>
        First line
      </p>
    </li>
    <li>
      <p>
        Second Line
      </p>
    </li>
    <li>
      <p>
        Third Line
      </p>
    </li>
  </ul>
</div>
`;

exports[`List component renders the unordered magenta list correctly 2`] = `
<div
  class="flex items-center"
  data-testid="list-element"
>
  <ol
    class="list-decimal marker:text-pink-800 marker:mr-2 marker:pr-2 space-y-2 marker:leading-snug"
  >
    <li>
      <p>
        First line
      </p>
    </li>
    <li>
      <p>
        Second Line
      </p>
    </li>
    <li>
      <p>
        Third Line
      </p>
    </li>
  </ol>
</div>
`;
